### ISSUE TITLE
what does the @ means in model.py?

### ISSUE THREAD

**Issue by qianchen94** [rosinality/sagan-pytorch#1]
with torch.no_grad():
            v = weight_mat.t() @ u
            v = v / v.norm()
            u = weight_mat @ v
            u = u / u.norm()
        sigma = u @ weight_mat @ v

**Comment by rosinality** [comment]
Hi, @ is matrix multiplication operator added in Python 3.5. You can replace it with torch.matmul.

**Comment by Zonobia-A** [comment]
Hello，i want to know that what are the real_image and label mean in programe,i don't know how real_image and label get from sample_data function.

**Comment by rosinality** [comment]
real_image and label corresponds to image of dataset and class label of that image. sample_data function yields this data by using ImageFolder dataset class. (in torchvision library) Please refer to https://pytorch.org/docs/stable/torchvision/datasets.html#imagefolder for explanations of ImageFolder class.

**Comment by Zonobia-A** [comment]
Thank you very much,can you tell me which dateset you choose for this programe?

**Comment by rosinality** [comment]
I have used subset of ILSVRC 2012. (10 classes)

**Comment by Zonobia-A** [comment]
Thanks!
